ddtrace/profiling/collector/stack.pyx

```diff
@@ -309,7 +309,11 @@ cdef stack_collect(ignore_profiler, thread_time, max_nframes, interval, wall_tim
     exc_events = []
 
     for thread_id, thread_native_id, thread_name, thread_pyframes, exception, span, cpu_time in running_threads:
-        thread_task_id, thread_task_name, thread_task_frame = _task.get_task(thread_id)
+        if thread_name is None:
```

Comment:
> does this create an observability gap for applications which pathologically spend a reasonable fraction of CPU-time in thread initialization?

The expectation is that this only happens when a thread is initialised. Under the assumption that an application doesn't spawn threads at very high frequency (it would not do good to Python performance), we would be dealing with long-lived threads on a normal basis, meaning that the initialisation CPU time would disappear in aggregates. So my very optimistic hunch is that we don't pay a huge price in terms of neglected data here.

> Is there any advantage to taking these threads, but giving them one-off names ("unnamed_thread#thread_id" or whatever)?

I think this is the current behaviour, and the picture looks like this

<img width="1050" alt="Screenshot 2023-03-20 at 16 19 41" src="https://user-images.githubusercontent.com/20231758/226403460-d57f05f3-9848-455c-a440-9bee2fb2f56c.png">

so no much useful initialisation data visible there. So probably still worth filtering here to declutter the thread list?